articles/user-guide/libraries/standard/diagnostics.md

```diff
@@ -30,20 +31,20 @@ Message($"About to rotate by an angle of {angle}...");
 
 The <xref:microsoft.quantum.diagnostics.dumpmachine> and <xref:microsoft.quantum.diagnostics.dumpregister> callables instruct target machines to provide diagnostic information about all currently allocated qubits or about a specific register of qubits, respectively.
 Each target machine varies in what diagnostic information is provided in response to a dump instruction.
-The [full state simulator](xref:microsoft.quantum.machines.full-state-simulator) target machine, for instance, provides the host program with the state vector that it uses internally to represent a register of qubits.
+The [full state simulator](xref:microsoft.quantum.machines.full-state-simulator) target machine, for example, provides the host program with the state vector that it uses internally to represent a register of qubits.
 By comparison, the [Toffoli simulator](xref:microsoft.quantum.machines.toffoli-simulator) target machine provides a single classical bit for each qubit.
 
- To learn more about the [full state simulator's](xref:microsoft.quantum.machines.full-state-simulator) `DumpMachine` output, take a look at the dump functions section of our [testing and debugging article](xref:microsoft.quantum.guide.testingdebugging#dump-functions).
+ To learn more about the `DumpMachine` output of the [full state simulator](xref:microsoft.quantum.machines.full-state-simulator), see **Dump functions** in [Testing and Debugging](xref:microsoft.quantum.guide.testingdebugging#dump-functions).
 
 
-## Facts and Assertions ##
+## Facts and assertions
 
 As discussed in [Testing and Debugging](xref:microsoft.quantum.guide.testingdebugging), a function or operation with signature `Unit -> Unit` or `Unit => Unit`, respectively, can be marked as a *unit test*.
 Each unit test generally consists of a small quantum program, along with one or more conditions that check the correctness of that program.
 These conditions can come in the form of either _facts_, which check the values of their inputs, or _assertions_, which check the states of one or more qubits passed as input.
 
-For example, `EqualityFactI(1 + 1, 2, "1 + 1 != 2")` represents the mathematical fact that $1 + 1 = 2$, while `AssertQubit(One, qubit)` represents the condition that measuring `qubit` will return a `One` with certainty.
-In the former case, we can check the correctness of the condition given only its values, while in the latter, we must know something about the state of the qubit in order to evaluate the assertion.
+For example, `EqualityFactI(1 + 1, 2, "1 + 1 != 2")` represents the mathematical fact that $1 + 1 = 2$, while `AssertQubit(One, qubit)` represents the condition that measuring `qubit` returns a `One` with certainty.
```

Comment:
The distinction between "returns" and "will return" is important here, such that the former is not actually correct. Assertions don't check the return value of a measurement operation (that would be a fact), but check the counterfactual condition that an _if_ an operation is run, then the return value will be correct, without actually running the given measurement operation.

Put differently, `EqualityFactR(M(qubit), One, "")` actually performs the measurement such that there is a concrete `Result` value to compare, while `AssertQubit(One, qubit)` does not perform a measurement at all but ensures that if a measurement were to be performed, it would always return `One` at that point.